### PR TITLE
draco: increase decompression performance

### PR DESCRIPTION
**Pull request by belom88**
Current master:
Screenshot from 2020-12-17 06-32-38
This branch:
Screenshot from 2020-12-17 06-29-51

I checked this on local loader (not worker).

### CODE REVIEW

**Comment by ibgreen** [comment]
This is very impressive - surprisingly so, as per the THREE.js issue I got the sense that the improvement should be around 20%, not 100%.

Any chance you can also test workers using the throughput measurement options we use for images?

There should be a way to reduce the number of iterations when setting up the Bench suite to make the benchmarks run faster.

File: modules/draco/src/lib/draco-parser.js [hunk 1]
```diff
@@ -217,19 +219,14 @@ export default class DracoParser {
...
-      const index = i * 3;
-      indices[index] = dracoArray.GetValue(0);
-      indices[index + 1] = dracoArray.GetValue(1);
-      indices[index + 2] = dracoArray.GetValue(2);
-    }
+    const byteLength = numIndices * INDEX_ITEM_SIZE;
+
+    const ptr = this.draco._malloc(byteLength);
+    decoder.GetTrianglesUInt32Array(dracoGeometry, byteLength, ptr);
+    const indices = new Uint32Array(this.draco.HEAPF32.buffer, ptr, numIndices).slice();
```
Comment (ibgreen) [inline modules/draco/src/lib/draco-parser.js:227]: Why do we `slice`? That creates a copy?

File: modules/draco/src/lib/draco-parser.js [hunk 2]
```diff
@@ -263,71 +260,54 @@ export default class DracoParser {
...
-        break;
+    const ptr = this.draco._malloc(byteLength);
+    decoder.GetAttributeDataArrayForAllPoints(
+      dracoGeometry,
+      dracoAttribute,
+      dataType,
+      byteLength,
+      ptr
+    );
+    const typedArray = new TypedArrayCtor(this.draco.HEAPF32.buffer, ptr, numValues).slice();
```
Comment (ibgreen) [inline modules/draco/src/lib/draco-parser.js:279]: Slice seems to create unnecessary copy?

**Review by ibgreen — APPROVED** [review]
Would be good to do some visual tests too. 
- Temporarily re-enable the render tests
- Test some DRACO compressed glTF models in `examples/website/gltf` (using `yarn start-local`)
- Test some glTF powered 3D Tiles modes in `examples/website/3d-tiles`.
-

**Comment by belom88** [comment]
I've done some testing with local worker: 
```
workerUrl: `modules/draco/dist/draco-loader.worker.dev.js`,
```
Current master:
Screenshot from 2020-12-18 14-41-44
This branch:
Screenshot from 2020-12-18 14-44-59
All visual tests pass.

File: modules/draco/src/lib/draco-parser.js (hunk 1 above)
Comment (belom88) [reply modules/draco/src/lib/draco-parser.js:227]: Wrapped `this.draco.HEAPF32.buffer` have to be disposed. A copy is necessary

File: modules/draco/src/lib/draco-parser.js (hunk 2 above)
Comment (belom88) [reply modules/draco/src/lib/draco-parser.js:279]: Wrapped `this.draco.HEAPF32.buffer` have to be disposed. A copy is necessary

**Pull request merged**